core/Controller/LostController.php

```diff
@@ -389,11 +392,11 @@ protected function findUserByIdOrMail($input) {
 			return $user;
 		}
 
-		$users = \array_filter($this->userManager->getByEmail($input), function (IUser $user) {
+		$users = array_values(array_filter($this->userManager->getByEmail($input), function (IUser $user) {
 			return $user->isEnabled();
-		});
+		}));
 
-		if (\count($users) === 1) {
+		if (count($users) === 1) {
 			return $users[0];
```

Comment:
could also use 
```suggestion
			return array_shift($users);
```

here instead of running array_values before.

Reply:
Fine by me. If the count is not 1 we save the array_values call and might be faster :racing_car:

Reply:
or simply use reset ? because we do not need the $users array anymore
```suggestion
			return reset($users);
```

Reply:
If I may , … a question 
is it still usefull to test element of returned array in IUserManager methods (getByEmail or get) ?
could it be simpler to remove this sort of tests:

`
               if ($user instanceof IUser) {
			return $user;
		}
`

Reply:
Sounds good to me. reset looks even faster than array_shift because it does not reindex the array: https://github.com/php/php-src/blob/57a385db09cb262ad2eb4191ac8fedb73e5df720/ext/standard/array.c